File: ruby/spec/static_array_spec.rb
require "static_array"

describe StaticArray do
  subject(:arr) { StaticArray.new }

  describe "#initialize" do
    it "starts with an empty and fixed default length of 8" do
      expect(arr.length).to eq(8)
      8.times { |i| expect(arr[i]).to be(nil)}
    end
    it "initializes diferent lengths" do
      expect(StaticArray.new(3).length).to eq(3)
    end
  end

  describe "#[] / #[]= " do
    it "sets and gets elements" do
      8.times { |i| arr[i] = i + 2 }
      8.times { |i| expect(arr[i]).to eq(i + 2)}
    end
  end

end

File: js/spec/StaticArraySpec.js
describe( "StaticArray", function() {
  let arr =  new StaticArray;
  describe( "#initialize", function() {
    it( "starts with an empty and fixed default length of 8", function() {
      expect( arr.length ).toEqual(8);
      for (let i = 0; i < 8; i++) {
        expect( arr[i] ).not.toBeDefined();
      }
  });

});
  describe( "#[] / #[]= ", function() {
    it( "sets and gets elements", function() {
      for (let i = 0; i < 8; i++) {
        arr[i] = i + 3;
      }
      for (let i = 0; i < 8; i++) {
        expect(arr[i]).toEqual( i + 3 );
      }
    });
  });
});
